src/widget/wtrackmenu.cpp

```diff
@@ -0,0 +1,199 @@
+#include "widget/wtrackmenu.h"
+
+#include <utility>
+#include <iostream>
+#include <library/trackmodel.h>
+#include <util/desktophelper.h>
+#include "widget/wlibrarytableview.h"
+#include "library/dao/trackdao.h"
+#include "library/trackcollection.h"
+
+
+WTrackMenu::WTrackMenu(QWidget *parent, TrackCollectionManager* pTrackCollectionManager)
+        : QMenu(parent),
+        m_pTrackCollectionManager(pTrackCollectionManager){
+    m_pLoadToMenu = new QMenu(this);
+    m_pLoadToMenu->setTitle(tr("Load to"));
+    m_pDeckMenu = new QMenu(this);
+    m_pDeckMenu->setTitle(tr("Deck"));
+    m_pSamplerMenu = new QMenu(this);
+    m_pSamplerMenu->setTitle(tr("Sampler"));
+
+    m_pPlaylistMenu = new QMenu(this);
+    m_pPlaylistMenu->setTitle(tr("Add to Playlist"));
+    connect(m_pPlaylistMenu, SIGNAL(aboutToShow()),
+            this, SLOT(slotPopulatePlaylistMenu()));
```

Comment:
update for Qt5 signal-slot syntax checked at compile time. Update all signal-slot connections in this new WTrackMenu class
```suggestion
    connect(m_pPlaylistMenu, &QMenu::aboutToShow,
            this, &WTrackMenu::slotPopulatePlaylistMenu);
```